src/app/components/DataLoader/DataLoader.jsx

```diff
@@ -1,22 +1,49 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import dataLoaderUtil from '@dataLoaderUtil';
+import {
+  updateLoadingStatus,
+  updateLastLoaded,
+} from '../../actions/Actions';
 
-class DataLoader extends React.Component {
+// The sole responsibility of the DataLoader is to trigger a data reload whenever
+// the location changes.
 
+class DataLoader extends React.Component {
   componentDidMount() {
-    dataLoaderUtil.loadDataForRoutes(this.props.location);
+    const { location, dispatch, lastLoaded } = this.props;
+    const {
+      search,
+      pathname,
+    } = location;
+    if (lastLoaded === `${pathname}${search}`) return dispatch(updateLoadingStatus(false));
+
+    return dataLoaderUtil.loadDataForRoutes(location, dispatch)
+      .then(() => {
+        dispatch(updateLastLoaded(`${pathname}${search}`));
```

Comment:
I think we probably still want this in `loadDataForRoutes` since we want to be able to not `updateLastLoaded` in case of redirect.